### DISCUSSION TITLE
trainer.fit() hangs if first defining a nn.Module and then passing it to pl.LightningModule

### DISCUSSION THREAD

**Discussion by zhiqiangdon** [Lightning-AI/pytorch-lightning#11055] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
This [tutorial](https://pytorch-lightning.readthedocs.io/en/latest/starter/style_guide.html) suggests using pl.LightningModule to define a system, i.e., first defining a nn.Module model and then passing it to pl.LightningModule like this:

```
class LitModel(LightningModule):
    def __init__(self, model: nn.Module = None):
        super().__init__()
        self.model = model
```

However, for the above case, I find trainer.fit() hangs after calling the `configure_optimizers`.  Nothing is shown in the terminal and GPUs are unused.

If I change the definition as follows: 

```
class LitModel(LightningModule):
    def __init__(self, config):
        super().__init__()
        self.model = MyModel(config)
```
The trainer.fit() works this time.

I am confused why the first definition causes trainer.fit() stuck, even the validation sanity check is not run.

**Comment by rohitgr7** [comment]
it should work fine. Mind sharing a reproducible example?

**Reply by zhiqiangdon** [reply]
Thanks @rohitgr7 ,

I have found the reason of hanging. It is due to that I called `self.save_hyperparameters()` without excluding the `model`.  After I do `self.save_hyperparameters(ignore=["model"])`, the trainer.fit() works.
